app/controllers/application_controller.rb

```diff
@@ -33,4 +35,42 @@ def audit_action
                          params:     params.except(:utf8, :_method, :authenticity_token, :controller, :action))
     end
   end
+
+  # Devise methods
+  def authenticate_user!
+    store_location_for(:registered_user, request.original_url)
+    store_location_for(:saml_user, request.original_url)
+    unless (saml_user_signed_in? || registered_user_signed_in?)
+      flash[:alert] = 'You need to sign in or sign up before continuing.'
```

Comment:
Users don't really sign up, do they? Maybe `"sign in or create an account"`?

Reply:
"sign up" is the devise language and the account creation comes from the sign up link. This can be changed but you need to go in and find every time that devise uses this language and override it.

Reply:
Will be resolved in #572